lib/max_subarray.rb

```diff
@@ -1,8 +1,32 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n!)? You only go throught the each with index n times, but the inner loop completes decrementing number of loops... 
+# Space Complexity: O(1), no additional data structures created outside of 2 variables each times
 def max_sub_array(nums)
```

Comment:
⚠️ I think this is O(n^2) due to the tested loop.  You can do it without the inner loop, and just with some if-else statements.